src/java.base/share/classes/java/util/Locale.java

```diff
@@ -980,33 +981,46 @@ static Locale getInstance(String language, String script, String country,
         return getInstance(baseloc, extensions);
     }
 
+
     static Locale getInstance(BaseLocale baseloc, LocaleExtensions extensions) {
         if (extensions == null) {
             Locale locale = CONSTANT_LOCALES.get(baseloc);
             if (locale != null) {
                 return locale;
             }
-            return LOCALE_CACHE.computeIfAbsent(baseloc, Locale::createLocale);
+            return LOCALE_CACHE.computeIfAbsent(baseloc, LOCALE_CREATOR);
         } else {
             LocaleKey key = new LocaleKey(baseloc, extensions);
-            return LOCALE_CACHE.computeIfAbsent(key, Locale::createLocale);
+            return LOCALE_CACHE.computeIfAbsent(key, LOCALE_CREATOR);
         }
     }
 
-    private static final ReferencedKeyMap<Object, Locale> LOCALE_CACHE = ReferencedKeyMap.create(true, ConcurrentHashMap::new);
-    private static Locale createLocale(Object key) {
-        if (key instanceof BaseLocale base) {
-            return new Locale(base, null);
+    private static final ReferencedKeyMap<Object, Locale> LOCALE_CACHE
+            = ReferencedKeyMap.create(true, ReferencedKeyMap.concurrentHashMapSupplier());
+
+    private static final Function<Object, Locale> LOCALE_CREATOR = new Function<>() {
+        @Override
+        public Locale apply(Object key) {
+            if (key instanceof BaseLocale base) {
+                return new Locale(base, null);
+            }
+            LocaleKey lk = (LocaleKey)key;
+            return new Locale(lk.base, lk.exts);
         }
-        LocaleKey lk = (LocaleKey)key;
-        return new Locale(lk.base, lk.exts);
-    }
+    };
 
     private static final class LocaleKey {
+
         private final BaseLocale base;
         private final LocaleExtensions exts;
         private final int hash;
 
+        private LocaleKey() {
+            base = null;
+            exts = null;
+            hash = 0;
+        }
```

Comment:
I don't think you need to add this constructor now.

Reply:
Was certain I had removed it.